nasa/harmony
# HARMONY-118: Simplify environment variables

Changed example/dotenv to a file that is sourced to set default environment variables and changed some of the defaults to be better suited for local development. The `.env.ts` file no longer has code to set default values reducing some complexity.

I also fixed the template generation script in this PR to fail if any of the parallel template generation fails.

I will add Bamboo variables to all the deployments for places where the defaults changed and we can no longer rely on the original defaults in deployed environments.

In total there are 5 variables needed in .env.

To test: 

1. Move off your existing .env file so that you have your original values saved.
2. Run npm test and verify that the main harmony tests and the query-cmr tests work without any .env file.
3. Run `bin/create-dotenv`
4. Set the 3 blank OAUTH parameters to match what was in your original .env
5. Add any additional overrides you want (for example if you don't usually run on port 3000)
6. Run `bin/start-argo` and verify there are no errors
7. Bring up harmony and make sure a request completes successfully.
